packages/gator-permissions-snap/src/permissions/erc20TokenRevocation/caveats.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  createValueLteTerms,
+  createAllowedCalldataTerms,
+  type Caveat,
+} from '@metamask/delegation-core';
+
+import type { PopulatedErc20TokenRevocationPermission } from './types';
+import type { DelegationContracts } from '../../core/chainMetadata';
+
+/**
+ * Appends permission-specific caveats for ERC20 approval revocation.
+ * @param args - The options object containing the permission and caveat builder.
+ * @param args.permission - The complete ERC20 token revocation permission containing revocation parameters.
+ * @param args.contracts - The contracts to use for the caveats.
+ * @returns The modified caveat builder with appended ERC20 token revocation caveats.
+ */
+export async function createPermissionCaveats({
+  contracts,
+}: {
+  permission: PopulatedErc20TokenRevocationPermission;
+  contracts: DelegationContracts;
+}): Promise<Caveat[]> {
+  // keccak("approve(address,uint256)"")
+  const approveFunctionSelector = '0x095ea7b3';
+  const zeroAsUInt256 =
+    '0x0000000000000000000000000000000000000000000000000000000000000000';
+
+  const functionSelectorCaveat: Caveat = {
```

Comment:
We also have `allowedMethodsEnforcer` Technically I don't think there is much difference using one or the other for this specific case. Readability wise it might be better with allowedMethod tho

Reply:
True - we don't have allowed methods in core though - so that would require another round of changes and release of the delegation-core package. I can create a ticket in the backlog.